Add the Chronicle Server chart (#441)
rstudio/helm: charts/posit-chronicle/README.md

# Posit Chronicle

![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![AppVersion: 2023.11.0](https://img.shields.io/badge/AppVersion-2023.11.0-informational?style=flat-square)

#### _Helm chart for the Chronicle Server_

> IT Administrators and Business Users use Posit Chronicle](https://www.rstudio.com/chronicle/) to aggregate and monitor

Reviewer: ```suggestion
IT Administrators and Business Users use [Posit Chronicle](https://www.rstudio.com/chronicle/) to aggregate and monitor
```

posit product usage.

## For Production

To ensure a stable production deployment, please:

* Ensure you "pin" the version of the Helm chart that you are using. You can do
  this using the `helm dependency` command and the associated "Chart.lock" files
  or the `--version` flag. **IMPORTANT: This protects you from breaking changes**
* Before upgrading, to avoid breaking changes, use `helm diff upgrade` to check
  for breaking changes
* Pay close attention to [`NEWS.md`](./NEWS.md) for updates on breaking
  changes, as well as documentation below on how to use the chart

## Installing the Chart

To install the chart with the release name `my-release` at version 0.1.0:

```bash
helm repo add rstudio https://helm.rstudio.com
helm upgrade --install my-release rstudio/posit-chronicle --version=0.1.0
```

To explore other chart versions, take a look at:
```
helm search repo rstudio/posit-chronicle -l
```

## Usage

> The Chronicle chart is meant to be used in tandem with other workbench and connect instances.
> To enable the chronicle agent, additional values will have to be passed to your
> workbench and connect values.
> 
> Here's some example agent helm values to run the agent sidecar in workbench:

Reviewer: ```suggestion
The Chronicle chart is meant to be used in tandem with other Workbench and Connect instances.
To enable the Chronicle agent, additional values will have to be passed to your
Workbench and Connect values.

Here's some example agent helm values to run the agent sidecar in Workbench:
```

```yaml
pod:
  ...
  sidecar:
    - name: chronicle-agent
      image: posit-chronicle:latest
      volumeMounts:
      - name: CHRONICLE_PRODUCT_CLUSTER_ID
        value: "posit-cluster-1"
      - name: logs
        mountPath: "/var/lib/rstudio-server/audit"
      env:
      - name: CHRONICLE_SERVER_ADDRESS
        value: "http://chronicle-server.default.svc.cluster.local"
...
```

> And here's some example agent helm values for connect, where we utilize an api key stored as a k8s secret
> to scrape the connect rest server:

Reviewer: ```suggestion
And here's some example agent helm values for Connect, where we utilize an API key stored as a k8s secret
to scrape the Connect metrics REST service:
```

```yaml
pod:
  ...
  sidecar:
    - name: chronicle-agent
      image: posit-chronicle:latest
      env:
      - name: CHRONICLE_PRODUCT_CLUSTER_ID
        value: "posit-cluster-1"
      - name: CHRONICLE_SERVER_ADDRESS
        value: "http://chronicle-server.default.svc.cluster.local"
      - name: CONNECT_API_KEY
        valueFrom:
          secretKeyRef:
            name: connect
            key: apikey
```

> Note that it will be up to the user to provision that connect api key Kubernetes secret.

Reviewer: ```suggestion
Note that it will be up to the user to provision that Connect API key Kubernetes secret.
```

## Storage Configuration

Chronicle can be configured to store its data in a local kubernetes volime, in S3,
or in both.

The default configuration uses a local volume, which is suitable if you'd like to
access and analyze the data within your cluster:

``` yaml
config:
  localStorage:
    enabled: true
    location: "./chronicle-data"
    retentionPeriod: "30d"
```

> RetentionPeriod accepts a duration string input. 0 implies infinite retention, disabling file expiration.

Reviewer: ```suggestion
`RetentionPeriod` accepts a duration string input. `0` implies infinite retention, disabling file expiration.
```

For example:

> 1s for 1 second, 5m for 5 minutes, 12h for 12 hours, 7d for one week, 365d for one year, 0 for unbound retention.

Reviewer: ```suggestion
`1s` for 1 second, `5m` for 5 minutes, `12h` for 12 hours, `7d` for one week, `365d` for one year, `0` for unbound retention.
```

and the server will store to both places. When using S3, you must also set the `S3Storage.Bucket`
parameter, like so:

``` yaml
config:
  s3Storage:
    enabled: true
    bucket: "posit-chronicle"
    prefix: ""
    profile: ""
    region: "us-east-2"
```

If you are running on EKS, we strongly suggest using [IAM Roles for Service
Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
to manage the credentials needed to access S3. In this scenario, once you have
[created an IAM
role](https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html),
you can use this role as an annotation on the existing Service Account:

``` yaml
serviceaccount:
  enabled: false
  # -- Additional annotations to add to the chronicle-server serviceaccount
  annotations: {
    eks.amazonaws.com/role-arn:  arn:aws:iam::123456789000:role/iam-role-name-here
  }
  # -- Additional labels to add to the chronicle-server serviceaccount
  labels: {}
```

If you are unable to use IAM Roles for Service Accounts, there are any number of
alternatives for injecting AWS credentials into a container. As a fallback, the

> s3 storage config allows specifying a profile:

Reviewer: ```suggestion
S3 storage config allows specifying a profile:
```

Author: wait i think the helm linter actually just reverted this update with a followup commit: https://github.com/rstudio/helm/pull/441/commits/70826ea346ad8a46e7eed13df8614999a5e6c658... hm..

Author: i think the linter was actually confused on some newline spacing, https://github.com/rstudio/helm/pull/441/commits/a8b77ea0f8291af66bb871442bab562f31ece826 should fix

Author: ah, i was editing this file directly not the template, go figure 😅.
Got it fixed now!
